### PR TITLE
Fix class instance handling

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**

### CODE REVIEW

File: lib/introspector.js [hunk 1]
```diff
@@ -338,13 +338,18 @@ const introspect = (namespace, text) => {
       const fn = iface[method];
       if (typeof fn !== 'function') {
         if (typeof fn === 'object') {
-          const isArray = Array.isArray(fn);
-          introspect(new Map([[method, fn]]), text).forEach(inv => {
-            inv.forEach((val, methodName) => {
-              if (isArray) entities.set(`${method}[${methodName}]`, val);
-              else entities.set(`${method}.${methodName}`, val);
+          const className = fn.constructor.name;
```
Comment (lundibundi) [inline lib/introspector.js]: This will crash for `fn === null` and `fn === Object.create(null)`

File: test/testConfig.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "files": ["test/example.js"],
   "customLinks": {
+    "ExampleClass": "#class-exampleclass",
```
Comment (lundibundi) [inline test/testConfig.json]: Could we remove this for now, and add support for automatic generation of class links in a separate PR? (where appropriate links will appear)

File: lib/introspector.js [hunk 3]
```diff
@@ -337,14 +337,24 @@ const introspect = (namespace, text) => {
...
-          introspect(new Map([[method, fn]]), text).forEach(inv => {
-            inv.forEach((val, methodName) => {
-              if (isArray) entities.set(`${method}[${methodName}]`, val);
-              else entities.set(`${method}.${methodName}`, val);
+        if (
+          typeof fn === 'object' &&
+          fn !== null &&
+          fn.constructor &&
+          fn.constructor.name
+        ) {
```
Comment (lundibundi) [inline lib/introspector.js]: Perhaps use `continue` here instead of additional indentation?

**Comment by SemenchenkoVitaliy** [comment]
ping @lundibundi

File: test/example.md [hunk 4]
```diff
@@ -233,7 +233,7 @@ ExampleClass constructor description
 
 method1 description
 
-## class AnotherClass extends ExampleClass
+## class AnotherClass extends `<ExampleClass>`
```
Comment (lundibundi) [inline test/example.md]: IMO the braces (`<>`) doesn't look good here, do we really need them?

**Comment by lundibundi** [comment]
Landed in 01c7061080c2fbed40854a6e9a71b9dbc2ba703f.

**Pull request closed without merging**